internal/server/http_params.go

```diff
@@ -0,0 +1,56 @@
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+)
+
+// HTTPParams contains all parameters for a remote lock request.
+type HTTPParams struct {
+	ClientParams Params `json:"client_params"`
+}
+
+// Params contains client parameters for a remote lock request.
+type Params struct {
+	NodeUUID string `json:"node_uuid"`
+	Group    string `json:"group"`
+}
+
+// NodeIdentity contains validated client identity from request parameters.
+type NodeIdentity struct {
+	UUID  string
+	Group string
+}
+
+func validateIdentity(req *http.Request) (*NodeIdentity, error) {
+	var group string
+	var nodeID string
+
+	if req.Header.Get("fleet-reboot") != "true" {
```

Comment:
I might be missing some context, is this just an arbitrary header that we want to ensure is present on requests so that we know we're talking to the right endpoint?

Reply:
Yes, it's a minor safety guard. The rationale is to prevent third-party requests from stealing reboot slots (e.g. on-host applications that allow POSTing body through them). Technique comes from https://cloud.google.com/compute/docs/storing-retrieving-metadata#querying and https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service#security.

Reply:
Fair, when I saw the header name I just wanted to make sure it was intentional rather than something we were accidentally absorbing from the old locksmith codebase.